### PR TITLE
Rename tracers to particles

### PR DESCRIPTION
**Pull request by gassmoeller**
The last in the series of PRs #1449 and #1451, and also a part of the changes planned in #1090
This is also incompatible to existing plugins and parameter files, but the rules in the update scripts are pretty simple and can be used to update parameter files and plugins.

### CODE REVIEW

**Comment by bangerth** [comment]
Some of the checks are failing. Can you take a look?

Also, would you mind squashing commits that only indent things with the commits that introduced the indentation errors?

**Comment by bangerth** [comment]
I think you also have a few unrelated commits in this PR.

File: include/aspect/boundary_fluid_pressure/density.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ namespace aspect
      * $\rho_s \textbf{g}$ (solid density times gravity) or to
      * $\rho_f \textbf{g}$ (fluid density times gravity).
      *
-     * @ingroup BoundaryFluidPressuressss
+     * @ingroup BoundaryFluidPressuresssss
```
Comment (bangerth) [inline include/aspect/boundary_fluid_pressure/density.h]: Nice try, but I noticed :-) There are others further down.

**Comment by gassmoeller** [comment]
aspect 8.5 succeeds so I will merge. I will open up a follow up PR for the regex issue (I fixed the expression already before this PR, but apparently I checked in the wrong modified files).

**Pull request merged**